kits/arm/example_teach_repeat_w_gripper.cpp

```diff
@@ -0,0 +1,245 @@
+  {
+    times[i] = (i+1) * wp_time;
+    target_pos.col(i) << state.waypoints[i].positions;
+    target_vels.col(i) << state.waypoints[i].vels;
+    target_accels.col(i) << state.waypoints[i].accels;
+  }
+  return arm::Goal::createFromWaypoints(times, target_pos, target_vels, target_accels);
+}
+
+
+double currentTime(std::chrono::steady_clock::time_point& start) {
+  return (std::chrono::duration<double>(std::chrono::steady_clock::now() - start)).count();
+}
+
+int main(int argc, char* argv[])
+{
+  //////////////////////////
+  ///// Arm Setup //////////
+  //////////////////////////
+
+  arm::Arm::Params params;
+
+  // Setup Module Family and Module Names
+  params.families_ = {"Arm"};
+  params.names_ = {"J1_base", "J2_shoulder", "J3_elbow", "J4_wrist1", "J5_wrist2", "J6_wrist3"};
+  
+  // Read HRDF file to setup a RobotModel object for the 6-DoF Arm
+  // Make sure you are running this from the correct directory!
+  params.hrdf_file_ = "kits/hrdf/6-dof_arm.hrdf";  
+
+  // Setup Gripper
+  std::shared_ptr<arm::EffortEndEffector<1>> gripper(arm::EffortEndEffector<1>::create(params.families_[0], "gripperSpool").release());
+  params.end_effector_ = gripper;
+  // auto end_effector = arm::EndEffector<1, arm::EndEffectorCommand::Effort>::create("Arm", "gripperSpool");
+  // Eigen::VectorXd aux_state(1);
+  // aux_state.setConstant(0);
+  // params_end_effector_ = end_gripper;
+
+
+
+  // Create the Arm Object
+  auto arm = arm::Arm::create(params);
+
+  /////////////////////////
+  //// MobileIO Setup /////
+  /////////////////////////
+
+  // Create the MobileIO object
+  std::unique_ptr<MobileIO> mobile = MobileIO::create("HEBI", "mobileIO");
+
+  // Clear any garbage on screen
+  mobile -> clearText(); 
+
+  // Setup instructions for display
+  std::string instructions;
+  instructions = ("B1 - Add stop WP\nB2 - Clear waypoints\n"
+                  "B3 - Add through WP\nB5 - Playback mode\n"
+                  "B6 - Grav comp mode\nB8 - Quit\n");
+
+  // Display instructions on screen
+  mobile -> sendText(instructions); 
+
+  // Setup state variable for mobile device
+  auto last_mobile_state = mobile->getState();
+
+
+  //////////////////////////
+  //// Main Control Loop ///
+  //////////////////////////
+
+  // Teach Repeat Variables
+  State state;
+  state.num_modules = arm -> robotModel().getDoFCount();
+
+  // Variable to switch gripper control from mobile_io to trajectory and vice versa
+  bool playback = false;
+
+  // Gripper Variables
+
+  while(arm->update())
+  {
+     // Get latest mobile_state
+    auto mobile_state = mobile->getState();
+    MobileIODiff diff(last_mobile_state, mobile_state);
+
+    // Testing
+    gripper->getState();
+    
+
+    double open_val = (mobile_state.getAxis(3) * 2)-1; // makes this range between -2 and 1
+    
+    if (!playback) {
+      // gripper -> setCommand(0, open_val);
+      gripper -> update(open_val);
+      gripper -> send();
+      // gripper 
```

Comment:
a couple of old comments?

Otherwise, looks good.  We can revisit the actual gripper logic details later.